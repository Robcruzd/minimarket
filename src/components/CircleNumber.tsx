import { StyleSheet, Text } from "react-native"

const CircleNumber = (props: {
    style?: {}; 
    quantity: number
}) => {
    return (
        <Text testID="circleNumber" style={[styles.textStyle, props.style]}>{props.quantity}</Text>
    )
}

export default CircleNumber;

const styles = StyleSheet.create({
    textStyle : {
        textAlign:"center",
        textAlignVertical:"center",
        zIndex: 2,
        backgroundColor:"#A51BB6", 
        color: "#fff",
        position:"absolute",
        width:30,
        height:30,
        borderRadius:50,
        top:5,
        left:5
        }
})