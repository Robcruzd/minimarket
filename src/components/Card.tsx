import { Dimensions, Image, StyleSheet, TouchableOpacity } from "react-native";
import CircleNumber from "./CircleNumber";

let widthCard = Dimensions.get("window").width/2 - 17;

const Card = (props: {myCart:any, product:any, navigation:any, index:number}) => {
    return (
        <TouchableOpacity 
            onPress={() => props.navigation.navigate("Product", { product: props.product, myCart:props.myCart})}
            style={styles.touch}
            key={props.index}>
                {props.myCart && 
                    <CircleNumber quantity={props.myCart.quantity}/>
                }
            <Image
                style={styles.imageProd}
                source={{
                uri: props.product.urlImage,
                }}
            />
        </TouchableOpacity>
    )
}

export default Card;

const styles = StyleSheet.create({
    touch: {
        borderWidth:1, 
        borderColor:"#aaaaaa", 
        margin: 5
    },
    imageProd: {
        width:widthCard, 
        height:widthCard
    }
})