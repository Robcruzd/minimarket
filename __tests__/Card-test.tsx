import React from 'react';
import { TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';
import Card from '../src/components/Card';
import CircleNumber from '../src/components/CircleNumber';

describe('<Card />', () => {
  const myCart = {
    quantity: 0,
    urlImage: 'https://example.com/image.png',
  };
  const product = {
      id: 1,
      quantity: 0,
      name: "",
      price: 1000,
      urlImage: 'https://example.com/image.png',
      description: "Lorem Ipsum is simply dummy text of the printing and typesetting industry."
    };
  const navigation = { navigate: jest.fn() };
  const index = 1;

  it('renders correctly', () => {
    const card = renderer
      .create(
        <Card myCart={myCart} product={product} navigation={navigation} index={index} />,
      )
      .toJSON();
    expect(card).toMatchSnapshot();
  });

  it('renders CircleNumber component if myCart prop is present', () => {
    const card = renderer.create(
      <Card myCart={myCart} product={product} navigation={navigation} index={index} />,
    );
    const instance = card.root;
    expect(instance.findByType(CircleNumber)).toBeTruthy();
  });

  it('does not render CircleNumber component if myCart prop is not present', () => {
    const card = renderer.create(
      <Card myCart={undefined} product={product} navigation={navigation} index={index} />,
    );
    const instance = card.root;
    expect(() => instance.findByType(CircleNumber)).toThrow();
  });

  it('navigates to Product screen when TouchableOpacity is pressed', () => {
    const card = renderer.create(
      <Card myCart={myCart} product={product} navigation={navigation} index={index} />,
    );
    const instance = card.root;
    instance.findByType(TouchableOpacity).props.onPress();
    expect(navigation.navigate).toHaveBeenCalledWith('Product', { product, myCart });
  });

});
