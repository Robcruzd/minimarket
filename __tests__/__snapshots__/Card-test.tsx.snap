// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderColor": "#aaaaaa",
      "borderWidth": 1,
      "margin": 5,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "backgroundColor": "#A51BB6",
          "borderRadius": 50,
          "color": "#fff",
          "height": 30,
          "left": 5,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 5,
          "width": 30,
          "zIndex": 2,
        },
        undefined,
      ]
    }
    testID="circleNumber"
  >
    0
  </Text>
  <Image
    source={
      {
        "uri": "https://example.com/image.png",
      }
    }
    style={
      {
        "height": 358,
        "width": 358,
      }
    }
  />
</View>
`;
